stock_available_to_promise_release/models/stock_move.py

```diff
@@ -83,7 +83,7 @@ def _previous_promised_qty_sql_lateral_where(self):
                 AND p_type.code = 'outgoing'
                 AND loc.parent_path LIKE ANY(%(location_paths)s)
                 AND (
-                    m.need_release = true
+                    COALESCE(m.need_release, False) = COALESCE(move.need_release, False)
```

Comment:
These fixes are normally authored by @jbaudoux from this PR: https://github.com/OCA/wms/pull/106